core/opendaq/device/src/device_info_impl.cpp

```diff
@@ -487,14 +504,24 @@ Int DeviceInfoConfigImpl<TInterface, Interfaces...>::getIntProperty(const String
     return obj.getPropertyValue(name).template asPtr<IInteger>();
 }
 
-OPENDAQ_DEFINE_CLASS_FACTORY_WITH_INTERFACE_AND_CREATEFUNC_OBJ(
-    LIBRARY_FACTORY,
-    DeviceInfoConfigImpl<>,
-    IDeviceInfoConfig,
-    createDeviceInfoConfig,
-    IString*,
-    name,
-    IString*,
-    connectionString)
+
+#if !defined(BUILDING_STATIC_LIBRARY)
+
+extern "C"
+ErrCode PUBLIC_EXPORT createDeviceInfoConfig(IDeviceInfoConfig** objTmp, IString* name, IString* connectionString)
```

Comment:
i dont see that we are using this in factory. do we need that?

Reply:
It's mostly here for binding generation. This factory should basically always be used, unless you know what you're doing. In the C++ wrapper we can use optional parameters, so we don't technically need this one.